modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Event.scala

```diff
@@ -28,13 +28,13 @@ object Event {
     * Events generated internally by the Engine.
     */
   sealed trait SystemEvent
-  case class Completed(i: Int) extends SystemEvent
-  case class Failed(i: Int) extends SystemEvent
+  case class Completed[R](i: Int, r: R) extends SystemEvent
+  case class Failed[E](i: Int, e: E) extends SystemEvent
   case object Executed extends SystemEvent
```

Comment:
It's just that I know something must go there, but I am not sure at this point what it is.